File: design_patterns/structural/facade/go/internal/computer.go
package computer

import "fmt"

type cpu struct{}

func (c *cpu) Execute() {
	fmt.Println("CPU: Executing instruction...")
}

func (c *cpu) Freeze() {
	fmt.Println("CPU: Freezing...")
}

type ram struct{}

func (m *ram) Load() {
	fmt.Println("RAM: Loading...")
}

func (m *ram) Flush() {
	fmt.Println("RAM: Flushing...")
}

type Computer struct {
	CPU *cpu
	RAM *ram
}

func (computer *Computer) Start() {
	fmt.Println("Starting computer!")
	computer.RAM.Load()
	computer.CPU.Execute()
	fmt.Println("Started!")
}

func (computer *Computer) Stop() {
	fmt.Println("Stoping computer!")
	computer.RAM.Flush()
	computer.CPU.Freeze()
	fmt.Println("Stopped!")
}

func NewComputer() *Computer {
	return &Computer{
		CPU: &cpu{},
		RAM: &ram{},
	}
}

File: design_patterns/structural/facade/python/src/facade/computer/computer.py
class RAM:
    def load(self) -> None:
        print("RAM: Loading...")

    def flush(self) -> None:
        print("RAM: Flushing...")


class CPU:
    def execute(self) -> None:
        print("CPU: Executing...")

    def freeze(self) -> None:
        print("CPU: Freezing...")


class Computer:
    def __init__(self) -> None:
        self._cpu = CPU()
        self._ram = RAM()

    def start(self) -> None:
        print("Computer: Starting!")
        self._ram.load()
        self._cpu.execute()
        print("Computer: Started!")

    def stop(self) -> None:
        print("Computer: Stopping!")
        self._cpu.freeze()
        self._ram.flush()
        print("Computer: Stopped!")
